src/lattice.cpp

```diff
@@ -710,69 +710,87 @@ const
 std::array<int, 3>
 HexLattice::get_indices(Position r, Direction u) const
 {
-  // The implementation for HexLattice currently doesn't use direction
-  // information. As a result, we move the position slightly forward to
-  // determine what lattice index the particle is most likely to be in.
-  r += TINY_BIT * u;
+  // result variables
+  int ix{}, ia{}, iz{};
 
   // Offset the xyz by the lattice center.
   Position r_o {r.x - center_.x, r.y - center_.y, r.z};
   if (is_3d_) {r_o.z -= center_.z;}
 
-  // Index the z direction.
-  std::array<int, 3> out;
+  // Index the z direction. Same as the technique used in
+  // RectLattice::get_indices.
```

Comment:
I would just say "accounting for coincidence" rather than making the reader go look at a different function.